src/common/components/telemetry-notice.tsx

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { ILinkProps } from 'office-ui-fabric-react';
+import * as React from 'react';
+import { NamedFC } from '../react/named-fc';
+
+export type LinkComponentDeps = {
+    LinkComponent: React.FC<ILinkProps>;
+};
```

Comment:
You're right to feel that it's strange to export `LinkComponentDeps` from `telemetry-notice.tsx`. I think this is a naming problem, rather than a location problem.

I would think of things this way: `TelemetryNotice` depends on `LinkComponent`. So `LinkComponent` should appear in `TelemetryNoticeDeps` (the set of tunneled dependencies of `TelemetryNotice`). In turn, a `deps` property of type `TelemetryNoticeDeps` should appear in the props type for `TelemetryNotice`, typically called `TelemetryNoticeProps`.

An example of this pattern is src\common\components\guidance-tags.tsx, where `GuidanceTags` depends on a `GetGuidanceTagsFromGuidanceLinks`.

So instead of calling the type above `LinkComponentDeps` I would call it `TelemetryNoticeDeps`. I'd define `TelemetryNoticeProps` that for now just contains `deps` of this type, and have that be the type parameter to `NamedFC` below.

Then this file exports `TelemetryNotice`, `TelemetryNoticeProps`, and `TelemetryNoticeDeps` which smells more coherent.

Reply:
This is great feedback. Thanks! 😄